components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java

```diff
@@ -908,7 +908,7 @@ public static APIRevisionDeployment mapAPIRevisionDeploymentWithValidation(Strin
             throw new APIManagementException("Invalid Vhost: " + vhost, ExceptionCodes.INVALID_VHOST);
         }
 
-        return mapApiRevisionDeployment(revisionId, vhost, displayOnDevportal, environment);
+        return mapApiRevisionDeployment(revisionId, vhost, displayOnDevportal, environment, environments.get(environment).getVisibility(), environments.get(environment).getPermissions().getPermissionType());
```

Comment:
This is longer than 120, shall we fix it,

Reply:
Done